src/components/Notification.tsx

```diff
@@ -1,23 +1,62 @@
-import { deleteNotificationFromDB } from "@/services/firebaseNotificationMethods";
-import { addUserToClientList } from "@/services/firebaseUserMethods";
 import { NotificationType } from "@/types/NotificationType";
+import { StateType } from "@/types/StateType";
+import { useSelector } from "react-redux";
+import { createToastNotification } from "@/helpers/createToastNotification";
+import { useDeleteNotificationFromListMutation } from "@/services/notifications";
+import { useAddClientToClientListMutation } from "@/services/clientLists";
 
-const handleDecline = (notificationId: string) => {
-  deleteNotificationFromDB(notificationId);
+const handleDecline = (onSuccess: () => void, onError: () => void) => {
+  try {
+    onSuccess();
+  } catch (error) {
+    console.log(error);
+    onError();
+  }
 };
 
-const handleAccept = (clientId: string, notificationId: string) => {
-  addUserToClientList(clientId)
-    .then(() => deleteNotificationFromDB(notificationId))
-    .catch((error) => console.log(error));
+const handleAccept = (onSuccess: () => void, onError: () => void) => {
+  try {
+    onSuccess();
+  } catch (error) {
+    console.log(error);
+    onError();
+  }
 };
 
 const Notification = ({ data }: { data: NotificationType }) => {
+  const user = useSelector((state: StateType) => state.userReducer.user);
+  const [deleteNotificationFromList] = useDeleteNotificationFromListMutation();
+  const [addClientToClientList] = useAddClientToClientListMutation();
+  const onSuccessDecline = () => {
+    deleteNotificationFromList({
+      userId: user.id,
+      notificationId: data.id,
+    });
+    createToastNotification("You declined colaboration");
+  };
+
+  const onSuccessAccept = () => {
+    addClientToClientList({ userId: user.id, clientId: data.from });
+    deleteNotificationFromList({
+      userId: user.id,
+      notificationId: data.id,
+    });
+    createToastNotification("You accepted colaboration");
+  };
+
+  const onError = () => {
+    createToastNotification("Something went wrong");
+  };
```

Comment:
I think this code might be added to mutations (config of them or something) or to the promise itself